lib/login-activity.js

```diff
@@ -0,0 +1,49 @@
+import { uuid, sparqlEscapeString, sparqlEscapeUri, sparqlEscapeDateTime } from 'mu';
+import { querySudo as query, updateSudo as update } from '@lblod/mu-auth-sudo';
+import { USERS_GRAPH, RESOURCE_BASE_URI } from '../config';
+
+const insertLoginActivity = async function(user) {
+  const now = new Date();
+  const queryResult = await query(`
+  PREFIX ext: <http://mu.semte.ch/vocabularies/ext/>
+  PREFIX prov: <http://www.w3.org/ns/prov#>
+
+  SELECT ?loginActivity
+  WHERE {
+    GRAPH ${sparqlEscapeUri(USERS_GRAPH)} {
+      ?loginActivity a ext:LoginActivity ;
+        prov:wasAssociatedWith ${sparqlEscapeUri(user)} .
+    }
+  }
+  LIMIT 1`);
+
+  if (queryResult.results.bindings.length) {
+    const loginActivity = queryResult.results.bindings[0].loginActivity.value;
+    await update(`
+    DELETE WHERE {
+      GRAPH ${sparqlEscapeUri(USERS_GRAPH)} {
+        ${sparqlEscapeUri(loginActivity)} ?p ?o .
+      }
+    }`);
+  }
```

Comment:
Just one `DELETE WHERE` query deleting all `ext:LoginActivity` associated with the user? No need to do the `SELECT` first